### ISSUE TITLE
Archiving fails when there is an analysis-archive delete transition

### ISSUE THREAD

**Issue by pkleimann** [anchore/anchore-engine#883]
**Is this a request for help?**:
No
---

**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Anchore Engine and Anchore CLI if applicable**:
Engine Code Version: 0.9.0

**What happened**:
If an analysis-archive rule with delete transition exists the archive process fails.

**What did you expect to happen**:
That the archive and delete process works without aborting with an error.

**Any relevant log output from /var/log/anchore**:
```
return self._evaluate_tag_history(rule, tag_histories_qry)
AttributeError: 'ArchivedAnalysisDeleter' object has no attribute '_evaluate_tag_history'
```

**Anything else we need to know:**
The error occurs here:
https://github.com/anchore/anchore-engine/blob/c045794dde9d282a6393c60febb4852fefe20353/anchore_engine/services/catalog/archiver.py#L742
I think it should be:
```
return self._evaluate_tag_history_and_exclude(rule, tag_histories_qry)
```

**Comment by dakaneye** [comment]
@pkleimann thanks for reporting! We made some improvements to the archive rules to support exclusion logic and a `max_images_per_account` setting, but we renamed this method without updating the child class as you pointed out. We'll get a PR opened ASAP.

**Comment by zhill** [comment]
Resolved by #884 and will be in v0.9.1 release
